### PR TITLE
feature/INTHAD-216-Map-export(EXPERIMENTAL)

### PR DESCRIPTION
**Pull request by shanmukhananda**

### CODE REVIEW

**Review by paul-michalik — COMMENTED** [review]
I suggest to use the ".vendor" strategy for copying/downloading (but not tracking) stuff from 3rd party libraries. This is now being integrated on the data source branch... Hm. let's keep this as it is for now.

File: experiments/cpp/slam-runner/main.cpp [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+#include <string>
+
+using namespace ORB_SLAM2;
+
+void dump_map(ext::map_export_t&& map) {
+    const char delim{' '};
+    const char nl{'\n'};
+    const int precision{15};
+
+    auto kfmap = std::get<ext::keyframe_map_t>(map);
```
Comment (paul-michalik) [inline experiments/cpp/slam-runner/main.cpp]: It is better to use `std::tie` before we get structured bindings:
```
std::tie(a, b, c) = map; 
```

**Comment by paul-michalik** [comment]
Hm. Looks like this is a mix of map export and covariance recovery, or am I mistaken?

**Comment by shanmukhananda** [comment]
Yes that's right, I have mixed map-export changes, cholmod solver cov recovery changes to this branch. I found it easier to keep all the changes in one place, as I was facing strange build issues across different platforms.

I would like to use this branch/PR to discuss the changes required to recover the co-variance, after this I will make separate discrete PRs.

**Comment by paul-michalik** [comment]
Ok, go ahead!

Sent from my Windows 10 phone

________________________________

**Pull request closed without merging**